File: ROS/ROS_Lab2/cpp_lab2/src/node2.cpp
// include ros client library and message types 
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
#include "std_msgs/msg/int64.hpp"
#include "cpp_lab2/msg/myinfo.hpp"


using std::placeholders::_1;

class MinimalSubscriber : public rclcpp::Node
{
  public:
    MinimalSubscriber()
    : Node("sub_node")
    {
       //create subscriber object
      subscription_ = this->create_subscription<cpp_lab2::msg::Myinfo>(
      "topic2", 10, std::bind(&MinimalSubscriber::topic_callback, this, _1));
       //create publisher object
      pub = this->create_publisher<std_msgs::msg::Int64>("int_fb", 10);
    }

  private:
    // define callback function
    void topic_callback(const cpp_lab2::msg::Myinfo::SharedPtr msg) const
    {
      RCLCPP_INFO(this->get_logger(), "Counter = '%d'", msg->num);
      // define message of type int
      auto x = std_msgs::msg::Int64();
      // fill fields of the message
      x.data = msg->num ;
      //publish the message
      pub->publish(x);

      
    }
    //decleration of subscriber 
    rclcpp::Subscription<cpp_lab2::msg::Myinfo>::SharedPtr subscription_;
    //decleration of publisher 
    rclcpp::Publisher<std_msgs::msg::Int64>::SharedPtr pub;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<MinimalSubscriber>());
  rclcpp::shutdown();
  return 0;
}

File: ROS/ROS_Lab_QOS/iti_lab4/iti_lab4/node2.py
#!/usr/bin/env python3

# 1) import ros client library 
import rclpy
from rclpy.node import Node
# 2) import message/service types 
from turtlesim.msg import Pose
 
#inherit from Node
class my_node(Node):
    def __init__(self):
        # node name  should be (unique)
        super().__init__("sub_node")
        # 3) create subscriber object (type,topic_name,callback,queue_size)
        self.create_subscription(Pose,"turtle1/custom_pose",self.callback,rclpy.qos.qos_profile_sensor_data  )
        
        self.get_logger().info("sub node is started now")

    # 4) define callback function called when recieving message from a topic
    def callback(self,msg):

        # 5) access message members msg.data
        #self.get_logger().info(f"{msg.x},{msg.y}")
        print(f"{msg.x},{msg.y}")
       



def main(args=None):
    # 6) start communication
    rclpy.init(args=args)
    # 7) create object from my node
    node = my_node()
    # keep the node running
    rclpy.spin(node)
    



    rclpy.shutdown()


if __name__ == "__main__":
    main()
# 8) don't forget to updte setup.py file


 